superset-frontend/src/dashboard/components/DrillDetailPane/DrillDetailPane.tsx

```diff
@@ -0,0 +1,239 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+  useRef,
+} from 'react';
+import {
+  BinaryQueryObjectFilterClause,
+  css,
+  ensureIsArray,
+  GenericDataType,
+  t,
+  useTheme,
+} from '@superset-ui/core';
+import Loading from 'src/components/Loading';
+import { EmptyStateMedium } from 'src/components/EmptyState';
+import TableView, { EmptyWrapperType } from 'src/components/TableView';
+import { useTableColumns } from 'src/explore/components/DataTableControl';
+import { getDatasourceSamples } from 'src/components/Chart/chartAction';
+import TableControls from './TableControls';
+
+type ResultsPage = {
+  total: number;
+  data: Record<string, any>[];
+  colNames: string[];
+  colTypes: GenericDataType[];
+};
+
+const PAGE_SIZE = 50;
+const MAX_CACHED_PAGES = 5;
+
+export default function DrillDetailPane({
+  datasource,
+  initialFilters,
+}: {
+  datasource: string;
+  initialFilters?: BinaryQueryObjectFilterClause[];
+}) {
+  const theme = useTheme();
+  const [pageIndex, setPageIndex] = useState(0);
```

Comment:
I'm using 0-indexed page indices in front-end code because the `TableView` component does and I felt like consistency with that was more important – that okay?

Reply:
The `TableView` has a property `initialPageIndex`, this property means which page is the first page when pagination. BTW, in the pagination context, the `index` means `page number` instead of `index`, so maybe the first page is 1.

https://github.com/apache/superset/blob/e214e1ace616c3fdd40fcf64c501e08407feb8b3/superset-frontend/src/components/TableView/TableView.tsx#L118-L139

Reply:
Oh great, didn't see that thanks!  Will update.

Reply:
@zhaoyongjie I'm having trouble using `initialPageIndex` to tell `TableView` to use 1-indexed page numbers: it seems like for server pagination, that parameter is used to set the current page number instead, and `TableView` expects the first page index to be `0`.  If you can figure out how to do it without rewriting `TableView` to expect 1-indexed page numbers, feel free to push a commit.